include/boost/ut.hpp

```diff
@@ -1127,9 +1131,9 @@ class reporter {
     printer_ << l.msg;
   }
 
-  auto on(events::exception) -> void {
+  auto on(events::exception exception) -> void {
     exception_ = true;
-    printer_ << "\n  " << printer_.colors().fail << "Unexpected exception!"
+    printer_ << "\n  " << printer_.colors().fail << exception.what()
```

Comment:
Good idea to add the exception.what() here - I also think it would be worth keeping the original "Unexpected exception" text here too... not all exception messages may convey that an exception had occurred.... Even better if the exception type could be included as well.

This may be what Catch prints (it may not print the exception type):
https://github.com/catchorg/Catch2/blob/dd3d27de5787da803e031b561196905e59e681bc/include/reporters/catch_reporter_tap.hpp#L102

Reply:
Actually if throw an exeption that doesn't override what(), it returns "Unknown exception", which seemed fine for me (similar enough to Unexpected exception!). But I changed that as well.